js/Utility.html

```diff
@@ -12,6 +12,77 @@
   array[index] = array[index].replace(/\s/g, '');
 },
 
+utility.stripItem = (value, index, array) => {
+  // only the quantity and barcode are needed, because the integrity
+  // check will pull the other information from the cache
+  array[index] = {
+    id: array[index].id,
+    description: array[index].description,
+    quantity: array[index].quantity,
+    barcode: array[index].barcode,
+  };
+},
+
+utility.itemIntegrityCheck = (item, index, array) => {
+  const items = app.cache.items;
+  const manualItemIdregex = /^MANUAL/;
+  var found = false;
+  // tries to locate the item in the inventory and
+  // sets the properties as they are in the inventory
+  for (let entry of items){
+    if (entry.barcode == item.barcode){
+      found = true;
+      item.description = entry.description;
+      item.serialized = entry.serialized;
+      if (item.serialized){
+        item.quantity = 1;
+      }
+      item.id = entry.id;
+    }
+  }
+  if (found || manualItemIdregex.test(item.id)){
+    item.checkedOut = true;
+    item.checkIn = null;
+    item.checkOut = null;
+    item.missing = false;
+    item.notes = '';
+  } else{
+    // removes the item if not manual and not found in inventory.
+    app.modal.handleError(new Error('Oops!, "' + item.description + '" was not found in our inventory. If this was a manual item, try entering it as a new manual item.'));
+    delete array[index];
```

Comment:
Does this work with the following processing of the array?
`delete` is going to make `array[index] == undefined`.  It does not remove that index from the array.  Is that OK?  What does the omnibox item handler do when it encounters an input of `undefined`?

Reply:
the omnibox item handler does nothing when it encounters undefined. I did array.splice(index,1) before and that did _not_ work because it would cause the array.foreach() that was calling this function to stop before the last item in the array.